src/main/scala/firrtl/annotations/JsonProtocol.scala

```diff
@@ -75,135 +105,238 @@ object JsonProtocol {
+          {
+            case tpe: Type => JString(tpe.serialize)
+          }
         )
       )
+
   class ExpressionSerializer
       extends CustomSerializer[Expression](format =>
         (
-          { case JString(s) => Parser.parseExpression(s) },
-          { case expr: Expression => JString(expr.serialize) }
+          {
+            case JString(s) => Parser.parseExpression(s)
+          },
+          {
+            case expr: Expression => JString(expr.serialize)
+          }
         )
       )
+
   class StatementSerializer
       extends CustomSerializer[Statement](format =>
         (
-          { case JString(s) => Parser.parseStatement(s) },
-          { case statement: Statement => JString(statement.serialize) }
+          {
+            case JString(s) => Parser.parseStatement(s)
+          },
+          {
+            case statement: Statement => JString(statement.serialize)
+          }
         )
       )
+
   class PortSerializer
       extends CustomSerializer[Port](format =>
         (
-          { case JString(s) => Parser.parsePort(s) },
-          { case port: Port => JString(port.serialize) }
+          {
+            case JString(s) => Parser.parsePort(s)
+          },
+          {
+            case port: Port => JString(port.serialize)
+          }
         )
       )
+
   class DefModuleSerializer
       extends CustomSerializer[DefModule](format =>
         (
-          { case JString(s) => Parser.parseDefModule(s) },
-          { case mod: DefModule => JString(mod.serialize) }
+          {
+            case JString(s) => Parser.parseDefModule(s)
+          },
+          {
+            case mod: DefModule => JString(mod.serialize)
+          }
         )
       )
+
   class CircuitSerializer
       extends CustomSerializer[Circuit](format =>
         (
-          { case JString(s) => Parser.parse(s) },
-          { case cir: Circuit => JString(cir.serialize) }
+          {
+            case JString(s) => Parser.parse(s)
+          },
+          {
+            case cir: Circuit => JString(cir.serialize)
+          }
         )
       )
+
   class InfoSerializer
       extends CustomSerializer[Info](format =>
         (
-          { case JString(s) => Parser.parseInfo(s) },
-          { case info: Info => JString(info.serialize) }
+          {
+            case JString(s) => Parser.parseInfo(s)
+          },
+          {
+            case info: Info => JString(info.serialize)
+          }
         )
       )
+
   class GroundTypeSerializer
       extends CustomSerializer[GroundType](format =>
         (
-          { case JString(s) => Parser.parseType(s).asInstanceOf[GroundType] },
-          { case tpe: GroundType => JString(tpe.serialize) }
+          {
+            case JString(s) => Parser.parseType(s).asInstanceOf[GroundType]
+          },
+          {
+            case tpe: GroundType => JString(tpe.serialize)
+          }
+        )
+      )
+
+  class UnrecognizedAnnotationSerializerSerializer
```

Comment:
Why does this have `Serializer` twice? Isn't it just an "UnrecognizedAnnotationSerializer"?

Reply:
If you're trying to say that it only serializes (and doesn't deserialize) I'd name it something like `UnrecognizedAnnotationSerializerOnly` or something.

Reply:
One serializer should be enough